### PR TITLE
Allow push jobs to build libzmq from source on windows

### LINKED ISSUE #125: Push jobs fails to install correctly
# Description

push jobs fails to install correctly. some dependencies seem to be missing so it cannot start the service and fails

# Push Jobs Client Version

2.2.0

# Platform Version

Windows Server 2008R2

# Replication Case

Method 1: use pushjobs cookbook to update the push jobs client to 2.2.0 > Result is pushjobs service fails to start

Method 2: uninstall manually old version and install version 2.2.0 from scratch > > Result is pushjobs service fails to start

Method 3: manually run installer for 2.2.0 without uninstalling old version first > Result is pushjobs service fails to start

# Stacktrace
```
[2017-04-04T14:39:12-04:00] FATAL: SystemCallError: windows_service[push-jobs-client] (push-jobs::service_windows line 6
3) had an error: SystemCallError: The service did not respond to the start or control request in a timely fashion. - Sta
rtService: The service did not respond to the start or control request in a timely fashion.


PS C:\Users\Administrator> c:\opscode\push-jobs-client\embedded\bin\ruby.exe c:\opscode\push-jobs-client\embedded\lib\ru
by\gems\2.3.0\gems\opscode-pushy-client-2.2.0\lib\pushy_client\windows_service.rb -c C:\chef/push-jobs-client.rb
Unable to load this gem. The libzmq library (or DLL) could not be found.
If this is a Windows platform, make sure libzmq.dll is on the PATH.
If the DLL was built with mingw, make sure the other two dependent DLLs,
libgcc_s_sjlj-1.dll and libstdc++6.dll, are also on the PATH.
For non-Windows platforms, make sure libzmq is located in this search path:
["C:/opscode/push-jobs-client/embedded/lib/ruby/gems/2.3.0/gems/ffi-rzmq-core-1.0.5/lib/ffi-rzmq-core/../../ext/libzmq.d
ll", "%SystemRoot%\\system32\\WindowsPowerShell\\v1.0\\/libzmq.dll", "c:\\Program Files\\7-Zip\\/libzmq.dll", "C:\\Windo
ws\\system32/libzmq.dll", "C:\\Windows/libzmq.dll", "C:\\Windows\\System32\\Wbem/libzmq.dll", "C:\\Windows\\System32\\Wi
ndowsPowerShell\\v1.0\\/libzmq.dll", "C:\\opscode\\chef\\bin\\/libzmq.dll", "C:/opscode/push-jobs-client/embedded/lib/li
bzmq.dll", "/usr/local/lib/libzmq.dll", "/opt/local/lib/libzmq.dll", "/usr/lib64/libzmq.dll"]
C:/opscode/push-jobs-client/embedded/lib/ruby/gems/2.3.0/gems/ffi-rzmq-core-1.0.5/lib/ffi-rzmq-core/libzmq.rb:57:in `res
cue in <module:LibZMQ>': The libzmq library (or DLL) could not be loaded (LoadError)
        from C:/opscode/push-jobs-client/embedded/lib/ruby/gems/2.3.0/gems/ffi-rzmq-core-1.0.5/lib/ffi-rzmq-core/libzmq.
rb:10:in `<module:LibZMQ>'
        from C:/opscode/push-jobs-client/embedded/lib/ruby/gems/2.3.0/gems/ffi-rzmq-core-1.0.5/lib/ffi-rzmq-core/libzmq.
rb:7:in `<top (required)>'
        from C:/opscode/push-jobs-client/embedded/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:68:in `re
quire'
        from C:/opscode/push-jobs-client/embedded/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:68:in `re
quire'
        from C:/opscode/push-jobs-client/embedded/lib/ruby/gems/2.3.0/gems/ffi-rzmq-core-1.0.5/lib/ffi-rzmq-core.rb:3:in
 `<top (required)>'
        from C:/opscode/push-jobs-client/embedded/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:68:in `re
quire'
        from C:/opscode/push-jobs-client/embedded/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:68:in `re
quire'
        from C:/opscode/push-jobs-client/embedded/lib/ruby/gems/2.3.0/gems/ffi-rzmq-2.0.4/lib/ffi-rzmq.rb:66:in `<top (r
equired)>'
        from C:/opscode/push-jobs-client/embedded/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:68:in `re
quire'
        from C:/opscode/push-jobs-client/embedded/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:68:in `re
quire'
        from c:/opscode/push-jobs-client/embedded/lib/ruby/gems/2.3.0/gems/opscode-pushy-client-2.2.0/lib/pushy_client/p
rotocol_handler.rb:18:in `<top (required)>'
        from c:/opscode/push-jobs-client/embedded/lib/ruby/gems/2.3.0/gems/opscode-pushy-client-2.2.0/lib/pushy_client.r
b:21:in `require_relative'
        from c:/opscode/push-jobs-client/embedded/lib/ruby/gems/2.3.0/gems/opscode-pushy-client-2.2.0/lib/pushy_client.r
b:21:in `<top (required)>'
        from c:/opscode/push-jobs-client/embedded/lib/ruby/gems/2.3.0/gems/opscode-pushy-client-2.2.0/lib/pushy_client/w
indows_service.rb:25:in `require_relative'
        from c:/opscode/push-jobs-client/embedded/lib/ruby/gems/2.3.0/gems/opscode-pushy-client-2.2.0/lib/pushy_client/w
indows_service.rb:25:in `<main>'
```

### PR DESCRIPTION
**Pull request by ksubrama**
### Description

See chef/omnibus-software#830 for details.  Don't merge until that goes in.

Also updates .kitchen.yml to build windows-2012r2 because we don't use win 2008 builders in the CI anymore.

### Issues Resolved

Resolves #125 (I hope :-P)

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [ ] CHANGELOG.md has been updated
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Comment by ksubrama** [comment]
@btm @charlesjohnson

**Comment by stevendanna** [comment]
Depnds on https://github.com/chef/omnibus-software/pull/830

File: omnibus/.kitchen.yml [hunk 1]
```diff
@@ -18,6 +18,13 @@ driver:
 provisioner:
   name: chef_zero
   require_chef_omnibus: true # Always install the latest version of Chef
+  attributes:
+    vagrant:
+      this_key_exists_so_we_have_a_vagrant_key: true
```
Comment (rhass) [inline omnibus/.kitchen.yml]: I am not entirely sure how this is used -- but Test Kitchen exports the shell environment variable `TEST_KITCHEN=1` as well.

File: omnibus/Gemfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 
 gem 'omnibus', git: 'https://github.com/chef/omnibus.git'
-gem 'omnibus-software', git: 'https://github.com/chef/omnibus-software.git'
+gem 'omnibus-software', git: 'https://github.com/chef/omnibus-software.git', branch: "ksubrama/con-640"
```
Comment (rhass) [inline omnibus/Gemfile]: I am having flashbacks of branch hell from before. Can we merge the omnibus PR?

Comment (stevendanna) [inline omnibus/Gemfile]: If we ever get it working, sure.

**Comment by stevendanna** [comment]
This is ready for review.  In a follow up commit we are going to try to get the service auto-starting again, but during testing I found that not starting the service was the only way to ensure we definitely installed completely.

**Comment by stevendanna** [comment]
"Proof" that the service can now start:

screen shot 2017-07-27 at 4 35 00 pm

Logs:

```
[2017-07-27T08:31:16-07:00] INFO: pushy-client is started...
[2017-07-27T08:31:16-07:00] INFO: [windows-test-node] Starting heartbeat / offline detection thread on interval 10.0 ...
[2017-07-27T08:31:16-07:00] DEBUG: [windows-test-node] Sending heartbeat (sequence #1)
[2017-07-27T08:31:16-07:00] INFO: [windows-test-node] Starting command / server heartbeat receive thread ...
[2017-07-27T08:31:16-07:00] INFO: [windows-test-node] Starting reconfigure thread.  Will reconfigure / reload keys after 3600 seconds, less up to splay 0.1.
[2017-07-27T08:31:16-07:00] INFO: [windows-test-node] Sending heartbeat 1 (logging first 3)
[2017-07-27T08:31:16-07:00] DEBUG: [windows-test-node] Received ZMQ message (Version:2.0;SigningMethod:hmac_sha256;Signature:8qlpb1c4KdKJ9jJfJPbnu3RBn07+HgBsb+yE5T+LBpA=, 73
[2017-07-27T08:31:16-07:00] INFO: [windows-test-node] Received abort
[2017-07-27T08:31:16-07:00] DEBUG: [] Job : command '' state 'idle'
[2017-07-27T08:31:16-07:00] DEBUG: [windows-test-node] Sending command aborted for job 
[2017-07-27T08:31:17-07:00] DEBUG: [windows-test-node] Received ZMQ message (Version:2.0;SigningMethod:rsa2048_sha1;Signature:mabHO2oeVPcInMuBOQFrbDqbZCvVowGM3x3iQHIMSru6bwdzjqbFGVLsQaMsd7fY/gojAAWa/IXgTTU36v2sO1qVW2tJXOuRKs0WAG+NLHuRSBKYbwYpOYaMR6KEnl9XvNpmlt3Y3Qdss/1uadiW9FcZzdPt5QT1X2KzW1RNvWpA9CzzYPif74Ee8NFX8bOR8vm0xLoOEVQwThw/fxsU4K69QWKhVjdkaWQX1RJYuzpdbH+eYmvi+bHxGDpYvof2ef287nneigbEPsNa+2PEtVo3sozZwKmnob4EjXpClx8SSZ6CpMWYjB8xdCwFrJyKgI9n0MH8iYnrWNVnQRMLGw==, 149
[2017-07-27T08:31:17-07:00] INFO: [windows-test-node] Received server heartbeat (sequence #59) logging 0/3
```

File: omnibus/.kitchen.yml (hunk 1 above)
Comment (jaym) [inline omnibus/.kitchen.yml]: I don't quite understand this

**Comment by jaym** [comment]
👍

**Pull request merged**